assets/terraform-modules/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -104,7 +99,6 @@ systemd:
           --read-only-port=0 \
           --register-with-taints=$${NODE_TAINTS} \
           --volume-plugin-dir=/var/lib/kubelet/volumeplugins
-        ExecStop=-/usr/bin/rkt stop --uuid-file=/var/cache/kubelet-pod.uuid
```

Comment:
Why there isn't a replacement for this line?

Reply:
I guess because old wrapper was running in the background, while `docker run` runs in the foreground and respects the signals for stopping?

Reply:
Makes sense. I'll wait for @surajssd confirmation if the kubelet stops correctly, etc. :)

Reply:
See the process is running:

```console
# systemctl status kubelet
● kubelet.service - Kubelet
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2020-09-16 12:44:45 UTC; 24min ago
  Process: 2877 ExecStartPre=/bin/mkdir -p /var/lib/kubelet/volumeplugins (code=exited, status=0/SUCCESS)
  Process: 2879 ExecStartPre=/usr/bin/bash -c grep 'certificate-authority-data' /etc/kubernetes/kubeconfig | awk '{print $2}' | base64 -d > /etc/kubernetes/c>
  Process: 2884 ExecStartPre=/etc/kubernetes/configure-kubelet-cgroup-driver (code=exited, status=0/SUCCESS)
...
```

Stopped the process using: `systemctl stop kubelet`. And it has stopped successfully:

```console
# systemctl status kubelet
● kubelet.service - Kubelet
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Wed 2020-09-16 13:09:50 UTC; 3s ago
  Process: 2877 ExecStartPre=/bin/mkdir -p /var/lib/kubelet/volumeplugins (code=exited, status=0/SUCCESS)
...
```

The docker container has stopped with code `0` 10seconds ago:

```console
# docker ps -a | grep kubelet
0b4165f8a945    quay.io/poseidon/kubelet:v1.18.8    "/usr/local/bin/kube…"   25 minutes ago     Exited (0) 10 seconds ago       elated_neumann
442a380563f2    quay.io/poseidon/kubelet:v1.18.8    "/usr/local/bin/kube…"   25 minutes ago     Exited (255) 25 minutes ago     cocky_keldysh
```

Reply:
Awesome, thanks! :)

Reply:
I am curious why [systemd-docker](https://github.com/ibuildthecloud/systemd-docker) is not needed anymore.

Reply:
Sorry, marked as resolved before seeing your comment, @alban 

Good question! @surajssd do you know? Following the issue linked in the README of the systemd-docker project, it is closed but only because it won't change in docker: https://github.com/moby/moby/issues/6791#issuecomment-411901825

I wasn't aware of that, but makes sense to investigate if those issues apply to us or not, IMHO. @surajssd what do you think?

Reply:
I think this works because the `docker run` is running in foreground. And once we do a `systemctl restart` only the client is killed effectively killing the service within. I will check the `systemd-docker` on Monday.

Reply:
I see that the code in system-docker is older than 5years. Should we use something more latest like podman maybe?

Reply:
This can also be useful: https://groups.google.com/g/flatcar-linux-user/c/IaB0NcA4780 (credits to alban for the link :))

Reply:
I have modified the code for AWS only right now. Now I use `ExecStartPre` to start the container in background with a predefined name. 

Also `ExecStart` only lists logs from the container and there are two more directives that are added `ExecStop` and `ExecStopPost` to stop and delete the kubelet container. PTAL if these changes look good to you folks.

It does not address the issue of moving the kubelet pid from one cgroup to another. But it does make sure that we don't have orphan kubelet container running, spawned by systemd.

Reply:
If these changes look good then I will add them to kubelet and etcd both.

Reply:
ping @alban @rata

Reply:
Alright I have went ahead and am resolving this conversation.

Reply:
Hi, sorry! I won't have time to follow-up on this or others, so please feel free to resolve them :)

To double check we are doing the right thing, I'd do something like the following:
 * Remove the kubelet daemonset
 * Make sure the bootstrap kubelet is running fine
 * `kill -9` the boostrap kubelet process (not the `docker run ...` command, but the kubelet process)
 * Check if the boostrap kubelet is restarted

I'm leaving this comment and running away for other things, so please ignore it if what I mention doesn't really make sense :D

Reply:
@rata yep checked all of those and the process restarts.